Tests/Service/Request/Visitor/RemoveNullFieldsRequestVisitorTest.php

```diff
@@ -0,0 +1,135 @@
+    public function testRemoveNullFields($input, $expected)
+    {
+        $this->requestProphecy->__call('withBody', [Argument::type(StreamInterface::class)])
+            ->shouldBeCalledOnce()
+            ->willReturn($this->requestProphecy);
+
+        $this->requestProphecy->__call('getMethod', [])
+            ->willReturn("PUT");
+
+        $this->requestProphecy->__call('getUri', [])
+            ->willReturn($this->uriInterfaceProfecy);
+
+        $this->uriInterfaceProfecy->__call('getPath', [])
+            ->willReturn("/v2/vehicles/43");
+
+        $this->streamInterfaceProfecy->__call('getContents', [])
+            ->willReturn($input);
+        $this->streamInterfaceProfecy->reveal();
+
+        $this->requestProphecy->__call('getBody', [])
+            ->willReturn($this->streamInterfaceProfecy);
+
+        $this->streamFactoryProfecy->__call('createStream', [Argument::is($expected)])
+            ->willReturn($this->streamInterfaceProfecy);
+
+        $request = $this->requestProphecy->reveal();
+        $factory = $this->streamFactoryProfecy->reveal();
+        $removeNullFieldsRequestVisitor = new RemoveNullFieldsRequestVisitor(
+            $factory,
+            [
+                new EndpointConfiguration("PUT", "/v2/vehicles/{id}")
+            ]
+        );
+
+        $removeNullFieldsRequestVisitor->visit($request);
+    }
+
+    public function provider()
+    {
+        return [
+            'remove first level null' => [
+                json_encode(["a" => 1, "b" => null]),
+                json_encode(["a" => 1])
+            ], 'remove sub level null' => [
+                json_encode(["a" => 1, "b" => [ "c" => 2, "d" => null]]),
+                json_encode(["a" => 1, "b" => [ "c" => 2 ]])
+            ],
+            'leaves key empty when all childs are null' => [
+                json_encode(["a" => 1, "b" => [ "c" => null, "d" => null]]),
+                json_encode(["a" => 1, "b" => []])
+            ],
+            'associative array' => [
+                json_encode(["test" =>  ["some", ["some" => null], 2]]),
+                json_encode(["test" =>  ["some", [], 2]])
+            ]
+        ];
+    }
+
+    public function testNotConfigureEndpoinsWillNotBeProcess()
+    {
+        $this->requestProphecy->__call('withBody', [Argument::type(StreamInterface::class)])
+            ->shouldNotBeCalled();
+
+        $this->requestProphecy->__call('getMethod', [])
+            ->willReturn("GET");
+
+        $this->requestProphecy->__call('getUri', [])
+            ->willReturn($this->uriInterfaceProfecy);
+
+        $this->uriInterfaceProfecy->__call('getPath', [])
+            ->willReturn("/v2/vehicles/43");
+
+        $request = $this->requestProphecy->reveal();
+        $factory = $this->streamFactoryProfecy->reveal();
+        $removeNullFieldsRequestVisitor = new RemoveNullFieldsRequestVisitor(
+            $factory,
+            [
+                new EndpointConfiguration("PUT", "/v2/vehicles/{id}")
```

Comment:
Public repository. Do not expose internal endpoints

Reply:
Done, changed to generic